File: JavaScript/busquedabinaria.js
const arreglo = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

console.log("---Busqueda binaria---");
console.log("El arreglo es: " + arreglo.join(" "));

const numeroBuscado = parseInt(prompt("Que numero quiere buscar?"));

let inicio = 0;
let final = arreglo.length - 1;
let encontrado = false;

while (inicio <= final) {
    const medio = Math.floor((inicio + final) / 2);

    if (arreglo[medio] === numeroBuscado) {
        console.log("El numero esta en el lugar: " + (medio + 1));
        encontrado = true;
        break;
    } else if (numeroBuscado < arreglo[medio]) {
        final = medio - 1;
    } else {
        inicio = medio + 1;
    }
}

if (!encontrado) {
    console.log("El numero no esta en el array");
}

File: BusquedaBinaria.java
import java.util.Scanner;

public class BusquedaBinaria {
    public static void main(String[] args) {
        int[] arreglo = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        Scanner sc = new Scanner(System.in);

        System.out.println("---Busqueda binaria---");
        System.out.print("El arreglo es: ");
        for (int i = 0; i < arreglo.length; i++) {
            System.out.print(arreglo[i] + " ");
        }

        System.out.print("\nQue numero quiere buscar? ");
        int numeroBuscado = sc.nextInt();

        int inicio = 0;
        int fin = arreglo.length - 1;

        while (inicio <= fin) {
            int medio = (inicio + fin) / 2;

            if (arreglo[medio] == numeroBuscado) {
                System.out.println("El numero esta en el lugar: " + (medio + 1));
                return;
            } else if (numeroBuscado < arreglo[medio]) {
                fin = medio - 1;
            } else {
                inicio = medio + 1;
            }
        }

        System.out.println("El numero no esta en el array");
    }
}